SyncExtension/UseCases/All/GetRemoteChangesUseCase.swift

```diff
@@ -200,4 +165,81 @@ class GetRemoteChangesUseCase {
+            debug: true
+        )
+        fileOffset += limit
+        let hasMoreFiles = updatedFiles.count == limit
+        
+        var folderDeletionCache: [String: Bool] = [:]
+
+        for file in updatedFiles {
+            let folderId = String(file.folderId)
+
+            if folderDeletionCache[folderId] == nil {
+                do {
+                    let folderMeta = try await APIFactory.DriveNew.getFolderMetaById(id: folderId)
+                    folderDeletionCache[folderId] = folderMeta.deleted
+                } catch {
+                    self.logger.error("❌ Error fetching folder metadata for file \(file.id): \(error.localizedDescription)")
+                    continue
+                }
+            }
+
+            if folderDeletionCache[folderId] == true {
+                deletedItemsIdentifiers.append(NSFileProviderItemIdentifier(rawValue: String(file.uuid)))
+                continue
+            }
+
+            if file.status == "REMOVED" || file.status == "TRASHED" {
+                deletedItemsIdentifiers.append(NSFileProviderItemIdentifier(rawValue: String(file.uuid)))
+                continue
+            }
+
+            if file.status == "EXISTS" {
+                guard let createdAt = Time.dateFromISOString(file.createdAt) else {
+                    self.logger.error("Cannot create createdAt date for item \(file.id) with value \(file.createdAt)")
+                    return
+                }
+                
+                guard let updatedAt = Time.dateFromISOString(file.updatedAt) else {
+                    self.logger.error("Cannot create updatedAt date for item \(file.id) with value \(file.updatedAt)")
+                    return
+                }
+                
+                if updatedAt > newFilesLastUpdatedAt {
+                    newFilesLastUpdatedAt = updatedAt
+                }
+                
+                let parentIsRoot = file.folderId == user.root_folder_id
+
+                let item = FileProviderItem(
+                    identifier: NSFileProviderItemIdentifier(rawValue: String(file.uuid)),
+                    filename: FileProviderItem.getFilename(name: file.plainName ?? file.name , itemExtension: file.type) ,
+                    parentId: parentIsRoot ? .rootContainer : NSFileProviderItemIdentifier(rawValue: file.folderId.toString()),
+                    createdAt: createdAt,
+                    updatedAt: updatedAt,
+                    itemExtension: file.type,
+                    itemType: .file,
+                    size: Int(file.size) ?? 0
+                )
+                
+                updatedFileProviderItems.append(item)
+            }
+        }
+        
+        if hasMoreFiles {
+            self.logger.info("There are more files, requesting them...")
+            try await self.obtainFileChanges(lastUpdatedAt: newFilesLastUpdatedAt, limit: self.enumeratedChangesLimit, recommendedBatchSize: recommendedBatchSize)
```

Comment:
On further updates, you should consider avoiding recursion when the process is potentially long-lived, like in this case to avoid stack overflowing.